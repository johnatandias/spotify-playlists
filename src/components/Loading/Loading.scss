.loading {
  animation: rotation infinite 5s linear;
}

@keyframes rotation {
  from { transform: rotate(0deg) }
  to { transform: rotate(360deg) }
}