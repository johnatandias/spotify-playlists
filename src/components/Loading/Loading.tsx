import React from 'react';
import './Loading.scss';

export const Loading = ({ width, height }: { width: number, height: number }) => (
  <svg viewBox="0 0 512.001 512.001" width={width} height={height} className="loading">
    <defs>
      <clipPath id="_clipPath_fskM7PJtNtaVAuu87nURuGGzdm3Ev1Ls">
        <rect width="512.001" height="512.001" />
      </clipPath>
    </defs>
    <g clipPath="url(#_clipPath_fskM7PJtNtaVAuu87nURuGGzdm3Ev1Ls)">
      <clipPath id="_clipPath_alg5W4kaaVWy8238tpAQgXzZf5BBbDnH">
        <rect x="0" y="0" width="512.001" height="512.001" transform="matrix(1,0,0,1,0,0)" fill="rgb(255,255,255)" />
      </clipPath>
      <g clipPath="url(#_clipPath_alg5W4kaaVWy8238tpAQgXzZf5BBbDnH)">
        <g>
          <g>
            <g>
              <path d=" M 10.407 187.915 C 22.137 145.439 44.899 106.426 76.232 75.092 C 124.656 26.669 189.041 0 257.524 0 C 326.007 0 390.391 26.669 438.816 75.094 L 445.427 81.705 L 445.427 19.991 C 445.427 8.95 454.377 0 465.418 0 C 476.459 0 485.409 8.95 485.409 19.991 L 485.409 93.112 C 485.33 108.951 479.168 123.87 468.056 135.123 C 456.905 146.415 442.03 152.739 426.17 152.93 L 354.469 152.931 C 343.428 152.931 334.478 143.981 334.478 132.94 C 334.478 121.899 343.428 112.949 354.469 112.949 L 420.129 112.949 L 410.544 103.364 C 369.671 62.491 315.327 39.981 257.524 39.981 C 199.721 39.981 145.377 62.491 104.504 103.364 C 78.051 129.817 58.84 162.735 48.947 198.558 C 46.5 207.421 38.457 213.233 29.689 213.232 C 27.928 213.232 26.136 212.997 24.355 212.505 C 13.713 209.568 7.467 198.558 10.407 187.915 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 301.666 468.85 C 307.595 467.559 313.542 465.996 319.339 464.205 C 329.887 460.947 341.081 466.854 344.341 477.404 C 347.601 487.953 341.691 499.146 331.143 502.406 C 324.263 504.531 317.208 506.385 310.173 507.918 C 308.739 508.23 307.31 508.38 305.901 508.38 C 296.704 508.38 288.424 501.992 286.388 492.637 C 284.038 481.848 290.878 471.198 301.666 468.85 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 384.242 431.233 C 389.115 427.608 393.895 423.727 398.449 419.702 C 406.722 412.388 419.356 413.169 426.666 421.44 C 433.979 429.712 433.2 442.347 424.929 449.658 C 419.537 454.424 413.878 459.017 408.109 463.31 C 404.525 465.977 400.34 467.264 396.191 467.264 C 390.088 467.264 384.06 464.478 380.136 459.205 C 373.545 450.346 375.383 437.823 384.242 431.233 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 444.161 362.977 C 447.125 357.668 449.893 352.173 452.39 346.642 C 456.935 336.58 468.777 332.107 478.838 336.651 C 488.9 341.195 493.373 353.037 488.829 363.098 C 485.867 369.655 482.584 376.171 479.07 382.464 C 475.411 389.021 468.608 392.714 461.597 392.714 C 458.299 392.714 454.955 391.896 451.871 390.175 C 442.231 384.796 438.779 372.618 444.161 362.977 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 470.654 276.173 C 471.16 270.135 471.417 263.981 471.417 257.884 C 471.417 246.843 480.367 237.893 491.408 237.893 C 502.449 237.893 511.399 246.843 511.399 257.884 C 511.399 265.091 511.095 272.368 510.496 279.512 C 509.621 289.946 500.883 297.835 490.598 297.835 C 490.037 297.835 489.473 297.812 488.905 297.764 C 477.902 296.842 469.731 287.175 470.654 276.173 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 100.054 436.43 C 106.583 427.528 119.092 425.6 127.998 432.129 C 132.901 435.724 138.015 439.151 143.194 442.312 C 152.619 448.064 155.595 460.367 149.842 469.791 C 146.074 475.963 139.498 479.37 132.76 479.37 C 129.21 479.37 125.615 478.424 122.364 476.439 C 116.224 472.692 110.165 468.632 104.356 464.372 C 95.452 457.844 93.526 445.334 100.054 436.43 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 18.784 259.616 C 29.796 258.621 39.504 266.718 40.504 277.714 C 41.055 283.767 41.87 289.865 42.929 295.838 C 44.855 306.709 37.603 317.084 26.732 319.011 C 25.552 319.22 24.379 319.321 23.221 319.321 C 13.708 319.321 5.277 312.506 3.56 302.814 C 2.306 295.735 1.339 288.509 0.686 281.338 C -0.315 270.342 7.789 260.617 18.784 259.616 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 40.355 356.782 C 49.956 351.331 62.158 354.699 67.609 364.3 C 70.604 369.576 73.864 374.792 77.299 379.804 C 83.541 388.91 81.22 401.353 72.113 407.596 C 68.658 409.965 64.723 411.099 60.828 411.099 C 54.458 411.099 48.195 408.059 44.321 402.409 C 40.25 396.47 36.387 390.287 32.837 384.034 C 27.387 374.433 30.753 362.232 40.355 356.782 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
          <g>
            <g>
              <path d=" M 187.172 484.618 C 189.445 473.811 200.048 466.896 210.85 469.168 C 216.779 470.415 222.846 471.423 228.881 472.166 C 239.84 473.513 247.632 483.488 246.284 494.447 C 245.039 504.577 236.42 512.001 226.468 512.001 C 225.655 512.001 224.831 511.951 224.004 511.85 C 216.848 510.97 209.654 509.775 202.621 508.296 C 191.816 506.024 184.9 495.422 187.172 484.618 Z " fill="rgb(26,177,106)" />
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
)