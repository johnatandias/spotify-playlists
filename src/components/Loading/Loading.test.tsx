import React from "react";
import { shallow } from 'enzyme';
import { Loading } from './Loading';

describe('Loading component', () => {
  it('Renders without crashing', () => {
    const size = { width: 300, height: 300 };
    const wrapper = shallow(<Loading {...size} />);
    expect(wrapper.find(`[height=${size.height}]`)).toHaveLength(1);
    expect(wrapper.find(`[width=${size.width}]`)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
