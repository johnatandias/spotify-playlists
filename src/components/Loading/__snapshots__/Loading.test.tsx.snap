// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component Renders without crashing 1`] = `ShallowWrapper {}`;
